### PR TITLE
Add the Danger gem to lint our PRs

### PR DESCRIPTION
**Pull request by lagartoflojo**
It will save us time by not having to make a human check some things.

### CODE REVIEW

**Comment by suse-tests-pass** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Warning">
          1 Warning
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false">Please include a CHANGELOG entry.<br />
Run <code>osc vc</code> to add one.</td>
    </tr>
  </tbody>
</table>

<p align="right" data-meta="generated_by_danger">
  Generated by :no_entry_sign: <a href="http://danger.systems/">Danger</a>
</p>

**Comment by suse-tests-pass** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Error">
          1 Error
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:no_entry_sign:</td>
      <td data-sticky="false"><table>
  <tbody>
    <tr>
      <td>app/models/product.rb</td>
      <td>57</td>
      <td>Lint/UselessAssignment: Useless assignment to variable - <code>hi</code>. (https://github.com/bbatsov/ruby-style-guide#underscore-unused-vars)</td>
    </tr>
  </tbody>
</table>
</td>
    </tr>
  </tbody>
</table>
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Warning">
          1 Warning
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false">Unless this is a trivial change, please include a CHANGELOG entry.<br />
Run <code>osc vc</code> to add one.</td>
    </tr>
  </tbody>
</table>

<p align="right" data-meta="generated_by_danger">
  Generated by :no_entry_sign: <a href="http://danger.systems/">Danger</a>
</p>

File: Dangerfile [hunk 1]
```diff
@@ -0,0 +1,46 @@
+# Flag that tells if any actual changes to the app were done
+has_app_changes = !git.modified_files.grep(/lib/).empty? || !git.modified_files.grep(/app/).empty?
+
+if !git.modified_files.include?('package/rmt-server.changes') && has_app_changes
+  warn("Unless this is a trivial change, please include a CHANGELOG entry.\nRun `osc vc` to add one.")
```
Comment (digitaltom) [inline Dangerfile]: -> Run `osc vc` in the package/ directory to add one.

File: Dangerfile [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+end
+
+# Run rubocop, failing if there's any violations
+rubocop.lint(report_danger: true, inline_comment: true)
+
+# Check that the versions in all files are the same
+def check_all_versions_equal
+  if makefile_version != spec_version || makefile_version != rmt_version || spec_version != rmt_version
+    error_msg = "The version of RMT is not consistent. These files should have the same version:\n"
+    error_msg << "`package/rmt-server.spec` specifies version `#{rmt_version}`\n"
```
Comment (digitaltom) [inline Dangerfile]: -> lib/rmt.rb

**Comment by lagartoflojo** [comment]
@digitaltom can you check again please?

**Pull request merged**